File: blaise-graphtheory3/src/test/java/com/googlecode/blaisemath/graph/mod/metrics/EigenCentralityTest.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package com.googlecode.blaisemath.graph.mod.metrics;

/*
 * #%L
 * BlaiseGraphTheory
 * --
 * Copyright (C) 2009 - 2019 Elisha Peterson
 * --
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import java.util.Arrays;
import java.util.Map;
import com.googlecode.blaisemath.graph.Graph;
import com.googlecode.blaisemath.graph.SparseGraph;
import static org.junit.Assert.*;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 *
 * @author elisha
 */
public class EigenCentralityTest {

    static Graph<Integer> TEST2;
    static EigenCentrality INST;

    @BeforeClass
    public static void setUpClass() throws Exception {
        System.out.println("-- EigenCentralityTest --");
        INST = new EigenCentrality();
        TEST2 = SparseGraph.createFromArrayEdges(false, Arrays.asList(1,2,3,4,5,6),
                Arrays.asList(
                    new Integer[]{1,2},
                    new Integer[]{1,3},
                    new Integer[]{2,3},
                    new Integer[]{2,6},
                    new Integer[]{3,4},
                    new Integer[]{4,5} ));
        //   1
        //  / \
        // 2---3--4--5
        // |
        // 6
        //
    }

    @Test
    public void testApply() {
        System.out.println("testApply");
        assertEquals(.475349771, INST.apply(TEST2, 1), 1e-8);
        assertEquals(.564129165, INST.apply(TEST2, 3), 1e-8);
        assertEquals(.296008301, INST.apply(TEST2, 4), 1e-8);
    }
    
    @Test
    public void testApply_All() {
        System.out.println("testApply_All");
        Map<Integer,Double> vals = INST.apply(TEST2);
        assertEquals(6, vals.size());
        for (int i = 0; i < 6; i++)
            assertEquals(INST.apply(TEST2, i+1), vals.get(i+1));
    }

}

File: blaise-graph-theory-kt/src/test/java/com/googlecode/blaisemath/graph/metrics/EigenCentralityTest.kt
package com.googlecode.blaisemath.graph.metrics

import com.google.common.graph.Graph
import com.googlecode.blaisemath.graph.GraphUtils
import com.googlecode.blaisemath.graphics.svg.SvgElementGraphicConverter
import com.googlecode.blaisemath.graphics.svg.SvgGraphic
import com.googlecode.blaisemath.graphics.svg.SvgGraphicComponent
import com.googlecode.blaisemath.graphics.svg.SvgUtils
import com.googlecode.blaisemath.style.xml.AttributeSetAdapter
import com.googlecode.blaisemath.svg.HelloWorldSvg
import com.googlecode.blaisemath.svg.SvgCircle
import com.googlecode.blaisemath.svg.SvgCircle.CircleConverter
import com.googlecode.blaisemath.svg.SvgElement
import com.googlecode.blaisemath.svg.SvgElements
import com.googlecode.blaisemath.svg.SvgEllipse
import com.googlecode.blaisemath.svg.SvgEllipse.EllipseConverter
import com.googlecode.blaisemath.svg.SvgGroup
import com.googlecode.blaisemath.svg.SvgImage
import com.googlecode.blaisemath.svg.SvgImage.ImageConverter
import com.googlecode.blaisemath.svg.SvgIo
import com.googlecode.blaisemath.svg.SvgLine
import com.googlecode.blaisemath.svg.SvgLine.LineConverter
import com.googlecode.blaisemath.svg.SvgNamespaceFilter
import com.googlecode.blaisemath.svg.SvgPath
import com.googlecode.blaisemath.svg.SvgPath.SvgPathOperator
import com.googlecode.blaisemath.svg.SvgPathTest
import com.googlecode.blaisemath.svg.SvgPolygon
import com.googlecode.blaisemath.svg.SvgPolygon.PolygonConverter
import com.googlecode.blaisemath.svg.SvgPolyline
import com.googlecode.blaisemath.svg.SvgPolyline.PolylineConverter
import com.googlecode.blaisemath.svg.SvgRectangle
import com.googlecode.blaisemath.svg.SvgRectangle.RectangleConverter
import com.googlecode.blaisemath.svg.SvgRoot
import com.googlecode.blaisemath.svg.SvgRootTest
import com.googlecode.blaisemath.svg.SvgText
import com.googlecode.blaisemath.svg.SvgText.TextConverter
import com.googlecode.blaisemath.svg.SvgTool
import com.googlecode.blaisemath.util.Images
import org.junit.Assert
import org.junit.BeforeClass
import org.junit.Test
import java.util.*

/*
* #%L
* BlaiseGraphTheory
* --
* Copyright (C) 2009 - 2021 Elisha Peterson
* --
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*      http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* #L%
*/   class EigenCentralityTest {
    @Test
    fun testApply() {
        Assert.assertEquals(.475349771, INST.apply(TEST2, 1), 1e-8)
        Assert.assertEquals(.564129165, INST.apply(TEST2, 3), 1e-8)
        Assert.assertEquals(.296008301, INST.apply(TEST2, 4), 1e-8)
    }

    @Test
    fun testApply_All() {
        val values = INST.apply(TEST2)
        Assert.assertEquals(6, values.size.toLong())
        for (i in 0..5) Assert.assertEquals(INST.apply(TEST2, i + 1), values[i + 1])
    }

    companion object {
        private var TEST2: Graph<Int?>? = null
        private var INST: EigenCentrality? = null
        @BeforeClass
        fun setUpClass() {
            INST = EigenCentrality()
            TEST2 = GraphUtils.createFromArrayEdges(false, Arrays.asList(1, 2, 3, 4, 5, 6),
                    Arrays.asList(arrayOf<Int?>(1, 2), arrayOf<Int?>(1, 3), arrayOf<Int?>(2, 3), arrayOf<Int?>(2, 6), arrayOf<Int?>(3, 4), arrayOf<Int?>(4, 5)))
            //   1
            //  / \
            // 2---3--4--5
            // |
            // 6
            //
        }
    }
}